service/history/mutableStateBuilder.go

```diff
@@ -1921,6 +1931,15 @@ func (e *mutableStateBuilder) addBinaryCheckSumIfNotExists(
 	exeInfo.AutoResetPoints = &workflow.ResetPoints{
 		Points: currResetPoints,
 	}
+	bytes, err := json.Marshal(recentBinaryChecksums)
```

Comment:
Can we add check: if ES is enabled then upsert binaryChecksums as search attributes?
Use `e.shard.GetConfig().AdvancedVisibilityWritingMode() != common.AdvancedVisibilityWritingModeOff` to check.

Reply:
Good call, done!